samplers/parser.go

```diff
@@ -389,23 +405,39 @@ func ParseMetric(packet []byte) (*UDPMetric, error) {
 			// should we be filtering known key tags from here?
 			// in order to prevent extremely high cardinality in the global stats?
 			// see worker.go line 273
-			tags := strings.Split(string(pipeSplitter.Chunk()[1:]), ",")
-			sort.Strings(tags)
-			for i, tag := range tags {
+			tagsList := strings.Split(string(pipeSplitter.Chunk()[1:]), ",")
+			sort.Strings(tagsList)
+			tags := make([]string, 0, len(tagsList))
+			for i, tag := range tagsList {
+				exclude := false
 				// we use this tag as an escape hatch for metrics that always
 				// want to be host-local
 				if strings.HasPrefix(tag, "veneurlocalonly") {
 					// delete the tag from the list
-					tags = append(tags[:i], tags[i+1:]...)
+					tagsList = append(tagsList[:i], tagsList[i+1:]...)
 					ret.Scope = LocalOnly
 					break
 				} else if strings.HasPrefix(tag, "veneurglobalonly") {
 					// delete the tag from the list
-					tags = append(tags[:i], tags[i+1:]...)
+					tagsList = append(tagsList[:i], tags[i+1:]...)
```

Comment:
you didn't change the second array name.
I think for easier review (and less rejects), I would keep the naming original naming convention and appending only the things you need... WDYT?
I got a little confused here, I'm not sure you're using the tagList anymore and I'm not sure it's good, there is a logic behind the tagList and the Scope